sherpa/include/sherpa/models.hh

```diff
@@ -24,9 +24,24 @@
 #include <sherpa/utils.hh>
 #include <sherpa/constants.hh>
 
+#include "Faddeeva.hh"
 
 namespace sherpa { namespace models {
 
+  template <typename DataType, typename ConstArrayType>
+  inline int wofz_point( const ConstArrayType& p, DataType x,
+                         DataType& val )
+  {
+    const DataType p0 = 2 * p[0];
+    const DataType p1 = 2 * p[1];
+    double sigma = p0 / std::sqrt(2 * log(2));
+    std::complex<double> arg = (x - p[2]) + p1 * std::complex<double>(0,1);
+    arg /= sigma * std::sqrt(2);
+    val = p[3] * Faddeeva::w(arg).real() / (sigma * std::sqrt(2 * PI));
+    return EXIT_SUCCESS;
+    
+  }
+    
```

Comment:
I found I had to change the interpretation of wofz as the scaling between fwhm_l and gamma was inverted:

```
@@ -32,16 +32,19 @@ namespace sherpa { namespace models {
   inline int wofz_point( const ConstArrayType& p, DataType x,
                          DataType& val )
   {
-    const DataType p0 = 2 * p[0];
-    const DataType p1 = 2 * p[1];
-    double sigma = p0 / std::sqrt(2 * log(2));
-    std::complex<double> arg = (x - p[2]) + p1 * std::complex<double>(0,1);
+    // Follow https://en.wikipedia.org/wiki/Voigt_profile after converting
+    // from FWHM.
+    const DataType fwhm_g = p[0];
+    const DataType fwhm_l = p[1];
+    const DataType gamma = fwhm_l / 2;
+    double sigma = fwhm_g / std::sqrt(8 * log(2));
+    std::complex<double> arg = (x - p[2]) + gamma * std::complex<double>(0,1);
     arg /= sigma * std::sqrt(2);
     val = p[3] * Faddeeva::w(arg).real() / (sigma * std::sqrt(2 * PI));
     return EXIT_SUCCESS;
-
+
   }
-
+
```

With this I now get a fit to the above dataset of 

```
In [4]: ui.fit('v')
Dataset               = v
Method                = levmar
Statistic             = leastsq
Initial fit statistic = 4329.82
Final fit statistic   = 10.9302 at function evaluation 313
Data points           = 401
Degrees of freedom    = 397
Change in statistic   = 4318.89
   vmdl.fwhm_g    2.10801      +/- 0.200975
   vmdl.fwhm_l    1.0508       +/- 0.224043
   vmdl.pos       9.24375      +/- 0.0266324
   vmdl.ampl      34.1915      +/- 1.08177

```

and this now matches the lmdif version as fwhm_l = 2 * gamma and fwhm_g = sigma / sqrt(8 * log(2)) 

It also now better matches the gauss1d and lorentz1d fits to the same data (the datasets are labelled by `g`, `l`, and `v` for gauss1d, lorentz1d, and voigt1d):

```
In [2]: ui.fit('g')
Dataset               = g
Method                = levmar
Statistic             = leastsq
Initial fit statistic = 4149.03
Final fit statistic   = 29.9943 at function evaluation 80
Data points           = 401
Degrees of freedom    = 398
Change in statistic   = 4119.04
   gmdl.fwhm      2.90164      +/- 0.0632627
   gmdl.pos       9.24277      +/- 0.0268652
   gmdl.ampl      9.81445      +/- 0.185295

In [3]: ui.fit('l')
Dataset               = l
Method                = levmar
Statistic             = leastsq
Initial fit statistic = 4213.59
Final fit statistic   = 53.7536 at function evaluation 101
Data points           = 401
Degrees of freedom    = 398
Change in statistic   = 4159.84
   lmdl.fwhm      2.30963      +/- 0.0716252
   lmdl.pos       9.24455      +/- 0.0252414
   lmdl.ampl      38.9724      +/- 0.854116

In [4]: ui.fit('v')
Dataset               = v
Method                = levmar
Statistic             = leastsq
Initial fit statistic = 4329.82
Final fit statistic   = 10.9302 at function evaluation 313
Data points           = 401
Degrees of freedom    = 397
Change in statistic   = 4318.89
   vmdl.fwhm_g    2.10801      +/- 0.200975
   vmdl.fwhm_l    1.0508       +/- 0.224043
   vmdl.pos       9.24375      +/- 0.0266324
   vmdl.ampl      34.1915      +/- 1.08177

```

So you can see the fwhm values of the voigt1d model now better match the gauss1d and lorentz1d fits.